### PR TITLE
Updating retriever-examples documentation to run validation tests on the provided snippets 

### PR DESCRIPTION
**Pull request by pmpailis**
Relates to https://github.com/elastic/elasticsearch/pull/116196 ; changing` js` snippets to `console` ones so that we can actually run validation for each provided query.

cc @leemthompo

### CODE REVIEW

File: docs/build.gradle [hunk 1]
```diff
@@ -1440,6 +1440,95 @@ setups['calendar_outages_addevent'] = setups['calendar_outages_addjob'] + '''
            ]}
 '''
 
+// Used by the retrievers-examples
+  buildRestTests.setups['retrievers_examples'] = '''
```
Comment (leemthompo) [inline docs/build.gradle:1444]: Normally these days we just do the test setup directly in the doc itself

see this [recent PR for example](https://github.com/elastic/elasticsearch/pull/115696/files#diff-2ed438e8b0249a1f0806ba278f579ae90da9e41a9b2135989dbbd3e55c37276dR107)

File: docs/reference/search/search-your-data/retrievers-examples.asciidoc [hunk 2]
```diff
@@ -425,4 +549,257 @@ GET retrievers_example/_search
...
+                                                {
+                                                    "value": 2.8129659,
+                                                    "description": "sum of:",
+                                                    "details": [
+                                                        {
+                                                            "value": 1.4064829,
+                                                            "description": "weight(text:information in 0) [PerFieldSimilarity], result of:",
+                                                            "details": [
+                                                                {
+                                                                    ...
```
Comment (pmpailis) [inline docs/reference/search/search-your-data/retrievers-examples.asciidoc:760]: This is TBD.. I wonder whether we should keep something simpler as the suggested output and ignore it from tests, or have more details (like in this case) and substitute where necessary.

**Comment by pmpailis** [comment]
@elasticmachine update branch

**Comment by pmpailis** [comment]
Will update to add an example using highlighting and inner-hits as well ( + update these references in the RRF retriever page).

File: docs/reference/search/rrf.asciidoc [hunk 3]
```diff
@@ -703,3 +703,89 @@ So for the same params as above, we would now have:
 
 * `from=0, size=2` would return [`1`, `5`] with ranks `[1, 2]`
 * `from=2, size=2` would return an empty result set as it would fall outside the available `rank_window_size` results.
+
+==== Aggregations in RRF
+
+Using the `rrf` retriever, we can also gather aggregations from all its specified sub-retrievers. The aggregations gathered
+are irrespective of the specified `rank_window_size` but instead we refer to the union of the result sets from all sub-retrievers,
+i.e. we collect all matching documents and not just the top `rank_window_size`.
```
Comment (leemthompo) [inline docs/reference/search/rrf.asciidoc:711]:
```suggestion
The `rrf` retriever supports aggregations from all specified sub-retrievers. Important notes about aggregations:

* They operate on the complete result set from all sub-retrievers
* They are not limited by the `rank_window_size` parameter
* They process the union of all matching documents
```

File: docs/reference/search/rrf.asciidoc [hunk 4]
```diff
@@ -703,3 +703,89 @@ So for the same params as above, we would now have:
...
+
+So for example, assuming that we have the following documents:
```
Comment (leemthompo) [inline docs/reference/search/rrf.asciidoc:713]:
```suggestion
For example, consider the following document set:
```

File: docs/reference/search/rrf.asciidoc [hunk 5]
```diff
@@ -703,3 +703,89 @@ So for the same params as above, we would now have:
...
+[source,python]
+----
+# doc     |   termA      |    termB    |
+_id: 1 =       foo
+_id: 2 =       foo             bar
+_id: 3 =     aardvark          bar
+_id: 4 =       foo             bar
+----
```
Comment (leemthompo) [inline docs/reference/search/rrf.asciidoc:721]:
```suggestion
[source,js]
----
{
    "_id": 1, "termA": "foo",
    "_id": 2, "termA": "foo", "termB": "bar",
    "_id": 3, "termA": "aardvark", "termB": "bar",
    "_id": 4, "termA": "foo", "termB": "bar"
}
----
```

Comment (leemthompo) [reply docs/reference/search/rrf.asciidoc:721]: maybe json makes more sense here?

File: docs/reference/search/rrf.asciidoc [hunk 6]
```diff
@@ -703,3 +703,89 @@ So for the same params as above, we would now have:
...
+// NOTCONSOLE
+
+And the following `rrf` query with a term aggregation specified on field `termA`:
```
Comment (leemthompo) [inline docs/reference/search/rrf.asciidoc:724]:
```suggestion
Perform a term aggregation on the `termA` field using an `rrf` retriever:
```

File: docs/reference/search/rrf.asciidoc [hunk 7]
```diff
@@ -703,3 +703,89 @@ So for the same params as above, we would now have:
...
+            "terms": {
+                "field": "termA"
+            }
+        }
+    }
+}
+----
+// NOTCONSOLE
+
+Even though we have a `rank_window_size: 1`, the aggregations will be computed against **all** matching documents from the nested sub-retrievers.
```
Comment (leemthompo) [inline docs/reference/search/rrf.asciidoc:763]:
```suggestion
The aggregation results will include *all* matching documents, regardless of `rank_window_size`.
```

File: docs/reference/search/rrf.asciidoc [hunk 8]
```diff
@@ -703,3 +703,89 @@ So for the same params as above, we would now have:
...
+So, the expected aggs would be:
```
Comment (leemthompo) [inline docs/reference/search/rrf.asciidoc:764]:
```suggestion

```

File: docs/reference/search/rrf.asciidoc [hunk 9]
```diff
@@ -703,3 +703,89 @@ So for the same params as above, we would now have:
...
+    "foo": 3,
+    "aardvark": 1
+}
+
+----
+// NOTCONSOLE
+
+==== Highlighting in RRF
+
+Similarly to above, we can also add <<highlighting, highlight snippets>> to the `rrf` retriever's results. Highlighted snippets are computed based
```
Comment (leemthompo) [inline docs/reference/search/rrf.asciidoc:777]:
```suggestion
Using the `rrf` retriever, you can add <<highlighting, highlight snippets>> to show relevant text snippets in your search results. Highlighted snippets are computed based
```

File: docs/reference/search/rrf.asciidoc [hunk 10]
```diff
@@ -703,3 +703,89 @@ So for the same params as above, we would now have:
...
+on the matching text queries defined on the sub-retrievers.
+
+NOTE:: Highlighting on vector fields, using either the `knn` retriever or a `knn` query, is not supported at the moment.
```
Comment (leemthompo) [inline docs/reference/search/rrf.asciidoc:780]:
```suggestion
IMPORTANT: Highlighting on vector fields, using either the `knn` retriever or a `knn` query, is not supported.
```

File: docs/reference/search/rrf.asciidoc [hunk 11]
```diff
@@ -703,3 +703,89 @@ So for the same params as above, we would now have:
...
+
+A more specific example of highlighting in RRF can also be found in the <<retrievers-examples-highlighting-retriever-results, retrievers examples>> page.
+
+==== Inner hits in RRF
+
+Computing <<inner-hits, inner hits>> is now also an option for RRF. We can specify inner hits as part of a nested sub-retriever, which
+will be propagated to the top level parent retriever. Actual inner hit computation will take place only at end of `rrf` retriever's evaluation on the top matching documents,
+and not as part of the query execution of the nested sub-retrievers.
```
Comment (leemthompo) [inline docs/reference/search/rrf.asciidoc:788]:
```suggestion
The `rrf` retriever supports <<inner-hits,inner hits>> functionality, allowing you to retrieve 
related nested or parent/child documents alongside your main search results. Inner hits can be 
specified as part of any nested sub-retriever and will be propagated to the top-level parent 
retriever. Note that the inner hit computation will take place only at end of `rrf` retriever's 
evaluation on the top matching documents, and not as part of the query execution of the nested 
sub-retrievers.
```

File: docs/reference/search/search-your-data/retrievers-examples.asciidoc [hunk 12]
```diff
@@ -94,10 +76,12 @@ POST /retrievers_example/_doc/5
...
 }
 
+POST /retrievers_example/_refresh
+
 ----
-//NOTCONSOLE
+// TESTSETUP
 
-Now that we also have our documents in place, let's try to run some queries using retrievers.
+Now that we  have our documents in place, let's try to run some queries using retrievers.
```
Comment (leemthompo) [inline docs/reference/search/search-your-data/retrievers-examples.asciidoc:84]:
```suggestion
Now that we have our documents in place, let's try to run some queries using retrievers.
```

File: docs/reference/search/search-your-data/retrievers-examples.asciidoc [hunk 13]
```diff
@@ -198,23 +223,875 @@ GET /retrievers_example/_search
...
        }
    },
-   "_source": ["text", "topic"]
+    "_source": false
+}
+----
+// TEST[continued]
+
+Which would return the following response collapsed results
+
```
Comment (leemthompo) [inline docs/reference/search/search-your-data/retrievers-examples.asciidoc:235]:
```suggestion
.Example response
[%collapsible]
==============
```

consider putting these responses inside collapsibles (nothing to do with "collapsed results 😛 )

File: docs/reference/search/search-your-data/retrievers-examples.asciidoc [hunk 14]
```diff
@@ -198,23 +223,875 @@ GET /retrievers_example/_search
...
+                        }
+                    }
+                }
+            }
+        ]
+    }
+}
+----
+// TESTRESPONSE[s/"took": 42/"took": $body.took/]
+
```
Comment (leemthompo) [inline docs/reference/search/search-your-data/retrievers-examples.asciidoc:329]:
```suggestion
==============
```

if you accept the suggestion above to use collapsible for responses :win

**Review by leemthompo — COMMENTED** [review]
Thanks @pmpailis  Just some cleanup suggestions, should be good to merge after this, will give a final look at preview :)

File: docs/reference/search/search-your-data/retrievers-examples.asciidoc (hunk 13 above)
Comment (pmpailis) [reply docs/reference/search/search-your-data/retrievers-examples.asciidoc:235]: yeap makes sense! will wrap all responses into collapsibles!

**Review by leemthompo — APPROVED** [review]
Just some final wording and formatting tweaks. Thanks @pmpailis!

I had one question about indentation in the retriever queries. Because they have so many parentheses, I would just double check that they're as left-aligned as possible :)

**Comment by pmpailis** [comment]
Thank you a lot @leemthompo  for all the iterations and the thorough review! :) 
Merged your suggestions, and will double check the indentation as suggested!

**Pull request merged**